pkg/controllers/management/oidcprovider/controller.go

```diff
@@ -179,20 +193,61 @@ func (c *oidcClientController) onChange(_ string, oidcClient *v3.OIDCClient) (*v
 		csids := strings.Split(clientSecretIDs, ",")
 		for _, csid := range csids {
 			delete(k8sSecret.Data, csid)
+			delete(k8sSecret.Annotations, clientSecretCreatedAtPrefixAnn+csid)
 		}
-		_, err = c.secretClient.Update(k8sSecret)
+		k8sSecret, err = c.secretClient.Update(k8sSecret)
 		if err != nil {
 			return nil, fmt.Errorf("failed to update secret: %w", err)
 		}
 
+		// delete annotation
 		delete(oidcClient.Annotations, removeClientSecretAnn)
-		_, err = c.oidcClient.Update(oidcClient)
+		oidcClient, err = c.oidcClient.Update(oidcClient)
 		if err != nil {
 			return nil, fmt.Errorf("failed to update OIDC client: %w", err)
 		}
 	}
 
-	return oidcClient, nil
+	return oidcClient, c.updateStatusIfNeeded(oidcClient, k8sSecret)
+}
+
+func (c *oidcClientController) updateStatusIfNeeded(oidcClient *v3.OIDCClient, secret *v1.Secret) error {
+	// calculate status
+	observedClientSecrets := map[string]v3.OIDCClientSecretStatus{}
+	for clientSecretName, clientSecretBytes := range secret.Data {
+		clientSecretValue := string(clientSecretBytes)
+		lastFiveCharacters := clientSecretValue
+		if len(clientSecretValue) > 5 {
+			lastFiveCharacters = clientSecretValue[len(clientSecretValue)-5:]
+		}
+		observedClientSecrets[clientSecretName] = v3.OIDCClientSecretStatus{
+			LastFiveCharacters: lastFiveCharacters,
+			CreatedAt:          secret.Annotations[clientSecretCreatedAtPrefixAnn+clientSecretName],
+			LastUsedAt:         oidcClient.Annotations[clientSecretUsedAtPrefixAnn+clientSecretName],
+		}
+	}
+	observedStatus := v3.OIDCClientStatus{
+		ClientID: secret.Name,
+	}
+	if len(observedClientSecrets) > 0 {
+		observedStatus.ClientSecrets = observedClientSecrets
+	}
+
+	if !reflect.DeepEqual(oidcClient.Status, observedStatus) {
+		patchData := map[string]interface{}{
+			"status": observedStatus,
+		}
+		patchBytes, err := json.Marshal(patchData)
+		if err != nil {
+			return fmt.Errorf("failed to create status patch: %w", err)
+		}
+		oidcClient, err = c.oidcClient.Patch(oidcClient.Name, types.MergePatchType, patchBytes, "status")
+		if err != nil {
+			return fmt.Errorf("failed to apply status patch: %w", err)
+		}
+	}
+
```

Comment:
Maybe we  split this into two parts?

If the `LastUsedAt` field can be concurrently updated, do we want to fail (and thus retry) because of a failed update of the timestamp?

Reply:
the annotation is updated in the token endpoint. Therefore, in the unlikely event of concurrent patches it will fail in the token endpoint not in the controller. I modified the code to only log this error, but don't return it to the user in https://github.com/rancher/rancher/pull/50069/commits/a53673ab0ae7ab66765ea0f532cac61895d6088b